pkg/apply/apply.go

```diff
@@ -42,12 +44,37 @@ func (a *Applicator) ApplyWorkflowWithHieraData(workflowName string, hieraData m
 
 func (a *Applicator) applyWithHieraData(workflowName string, hieraData map[string]string, intent wfapi.Operation) {
 	tp := func(ic lookup.ProviderContext, key string, _ map[string]eval.Value) (eval.Value, bool) {
-		v, ok := hieraData[key]
-		return types.WrapString(v), ok
+		m := convertToDeepMap(hieraData)
+		hclog.Default().Debug("converted map to hiera data", "m", m)
+		v := eval.Wrap(nil, m).(eval.OrderedMap)
+		return v, true
```

Comment:
This will return the whole map. You need `return v.Get4(key)` here.

Reply:
Change made.